Modules/_heapqmodule.c

```diff
@@ -279,16 +280,10 @@ _heapq_heappushpop_impl(PyObject *module, PyObject *heap, PyObject *item)
     return returnitem;
 }
 
-static Py_ssize_t
+static inline Py_ssize_t
 keep_top_bit(Py_ssize_t n)
 {
-    int i = 0;
-
-    while (n > 1) {
-        n >>= 1;
-        i++;
-    }
-    return n << i;
+    return (Py_ssize_t)1 << (_Py_bit_length((unsigned long)n) - 1);
```

Comment:
`_Py_bit_length(unsigned long x)` takes an unsigned long, whereas you have a `Py_ssize_t`. On Windows, `long` is only 32-bit, whereas size_t is 64-bit. Your cast is unsafe.

If you want to use _Py_bit_length(), you should add a 64-bit variant using `__builtin_clzll()` and `_BitScanReverse64()`.